R/epichains.R

```diff
@@ -55,19 +55,21 @@ new_epichains <- function(tree_df,
 #' @author James M. Azam
 #' @export
 epichains <- function(tree_df,
-                           index_cases,
-                           statistic,
-                           offspring_dist,
-                           stat_max,
-                           track_pop) {
+                      index_cases,
+                      statistic,
+                      offspring_dist,
+                      track_pop,
+                      stat_max = Inf) {
   # Check that inputs are well specified
-  checkmate::assert_data_frame(tree_df)
-  checkmate::assert_integerish(index_cases, null.ok = TRUE)
-  checkmate::assert_character(statistic, null.ok = TRUE)
+  checkmate::assert_data_frame(tree_df, min.cols = 3, min.rows = index_cases)
```

Comment:
Just being very nit-picky: would be worth checking the input column names and types? If this is primarily called internally from other {epichains} functions this might be less necessary, but could still be a good check to have.

Reply:
I check these in the `validate_*()` functions.